osu.Game.Tests/Visual/Online/TestSceneOnlineViewContainer.cs

```diff
@@ -0,0 +1,93 @@
+    public class TestSceneOnlineViewContainer : OsuTestScene
+    {
+        private readonly TestOnlineViewContainer onlineView;
+
+        public TestSceneOnlineViewContainer()
+        {
+            Child = new Container
+            {
+                RelativeSizeAxes = Axes.Both,
+                Child = onlineView = new TestOnlineViewContainer()
+            };
+        }
+
+        private class TestOnlineViewContainer : OnlineViewContainer
+        {
+            public new LoadingAnimation LoadingAnimation => base.LoadingAnimation;
+
+            public TestOnlineViewContainer()
+                : base(@"Please sign in to view dummy test content")
+            {
+                RelativeSizeAxes = Axes.Both;
+                Children = new Drawable[]
+                {
+                    new Box
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        Colour = Color4.Blue.Opacity(0.8f),
+                    },
+                    new OsuSpriteText
+                    {
+                        Text = "dummy online content",
+                        RelativeSizeAxes = Axes.Both,
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                    }
+                };
+            }
+
+            protected override void FadeContentOut(Drawable content) => content.Hide();
+
+            protected override void FadeContentIn(Drawable content) => content.Show();
+        }
+
+        [Test]
+        public void TestOnlineStateVisibility()
+        {
+            AddStep("set status to online", () => ((DummyAPIAccess)API).State = APIState.Online);
+
+            AddAssert("children are visible", () => onlineView.TransformationTarget.IsPresent);
+            AddAssert("loading animation is not visible", () => !onlineView.LoadingAnimation.IsPresent);
+        }
+
+        [Test]
+        public void TestOfflineStateVisibility()
+        {
+            AddStep("set status to offline", () => ((DummyAPIAccess)API).State = APIState.Offline);
+
+            AddAssert("children are hidden", () => !onlineView.TransformationTarget.IsPresent);
+            AddAssert("loading animation is not visible", () => !onlineView.LoadingAnimation.IsPresent);
+        }
+
+        [Test]
+        public void TestConnectingStateVisibility()
+        {
+            AddStep("set status to connecting", () => ((DummyAPIAccess)API).State = APIState.Connecting);
+
+            AddAssert("children are hidden", () => !onlineView.TransformationTarget.IsPresent);
+            AddAssert("loading animation is visible", () => onlineView.LoadingAnimation.IsPresent);
+
+            AddStep("set status to failing", () => ((DummyAPIAccess)API).State = APIState.Failing);
+
+            AddAssert("children are hidden", () => !onlineView.TransformationTarget.IsPresent);
+            AddAssert("loading animation is visible", () => onlineView.LoadingAnimation.IsPresent);
+        }
```

Comment:
I'd split this testcase in two (`TestConnectingStateVisibility` and `TestFailingStateVisibility`). You could even make all four test cases into a parameterised test for brevity but I guess that's just extra points.